docs/search_syntax.md

```diff
@@ -0,0 +1,228 @@
+# Search Syntax Documentation
+
+## Introduction
```

Comment:
what do you think about to have a `glossary` with links so you can easily jump to any section? not a requirement but just a thought if I want to quickly check special section.

Reply:
Yes! I'll add it